cmd/deploy.go

```diff
@@ -267,8 +275,15 @@ func runDeploy(cmd *cobra.Command, _ []string, newClient ClientFactory) (err err
 		return ErrRegistryRequired
 	}
 
+	// Load global config to persist the --remote flag
+	gc, err := fc.NewDefault()
+	if err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "error loading config at '%v'. %v\n", fc.ConfigPath(), err)
+	}
+	gc.Remote = cfg.Remote
+
```

Comment:
Maybe I am mis-reading, but this stanza _should_ not be necessary, because `cfg.Remote` should have already taken into account the static default (false), global config setting (via the flag default value), environment variables and flags (during the construction of the cfg struct)

The default value of the flag --remote will be set to either the static default (false) or the value defined in ~/.config/func/config.yaml`.  If the user decided to (globally) prefer remote, then `cfg.Remote` should reflect this value.

`cfg.Remote` also reflects if the user chose to override their global setting by explicitly providing a --remote flag (or environment variable).